### ISSUE TITLE
Media/relation inside a repeatable component disapears when saving

### ISSUE THREAD

**Issue by willianrod** [strapi/strapi#11955]
## Video demo
Recorded by @[dogunbound](https://github.com/dogunbound)
https://www.youtube.com/watch?v=j-TYly0s0L0

### Describe the bug
I'm using Strapi to build a customizable landing page with dynamic sections.

My current setup is this:

I have a component called `SkillItem`
<details>
  <summary>➡️ <strong>src/components/utils/skill-item.json</strong></summary>

```json
{
  "collectionName": "components_utils_skill_items",
  "info": {
    "displayName": "SkillItem",
    "icon": "server"
  },
  "options": {},
  "attributes": {
    "name": {
      "type": "string"
    },
    "icon": {
      "allowedTypes": [
        "images"
      ],
      "type": "media",
      "multiple": false,
      "required": true
    },
    "since": {
      "type": "date",
      "required": true
    }
  }
}
```
</details>

I have another component called `Skills` with a `items` field with a repeatable `SkillItem` component 

<details>
  <summary>➡️ <strong>src/components/sections/skills.json</strong></summary>
  
```json
{
  "collectionName": "components_sections_skills",
  "info": {
    "displayName": "Skills",
    "icon": "brain",
    "description": ""
  },
  "options": {},
  "attributes": {
    "title": {
      "type": "string",
      "required": true
    },
    "items": {
      "type": "component",
      "repeatable": true,
      "component": "utils.skill-item",
      "required": true
    }
  }
}
```
</details>

And I have a `Collection Type` called `Page` where I have a `Dynamic Zone` with all my sections components.

<details>
  <summary>➡️ <strong>src/api/page/content-types/page/schema.json</strong></summary>
  
```json
{
  "kind": "collectionType",
  "collectionName": "pages",
  "info": {
    "singularName": "page",
    "pluralName": "pages",
    "displayName": "Page",
    "description": ""
  },
  "options": {
    "draftAndPublish": true
  },
  "pluginOptions": {
    "i18n": {
      "localized": true
    }
  },
  "attributes": {
    "title": {
      "pluginOptions": {
        "i18n": {
          "localized": true
        }
      },
      "type": "string",
      "required": true
    },
    "slug": {
      "pluginOptions": {
        "i18n": {
          "localized": false
        }
      },
      "type": "string",
      "required": true,
      "unique": false
    },
    "description": {
      "pluginOptions": {
        "i18n": {
          "localized": true
        }
      },
      "type": "text"
    },
    "sections": {
      "pluginOptions": {
        "i18n": {
          "localized": true
        }
      },
      "type": "dynamiczone",
      "components": [
        "sections.welcome",
        "sections.about-me",
        "sections.work-experience",
        "sections.skills",
        "sections.goals"
      ]
    }
  }
}
```
</details>

When I'm editing my page and I have any items inside my skills section and I click on Save, my image disapears blocking me to publish my page, because strapi only allow me to publish if all required fields is filled.

If I edit an existing page and add a `Skills` section with an image inside an item, then save, then leave the page and come back to edit again, the image field is empty althouth this field has an value in my database.

### Steps to reproduce the behavior

![recording](https://user-images.githubusercontent.com/6539258/146797377-ede1fd8c-ec67-4c13-806a-649bde329cf7.gif)

1. Go to `Page`
2. Create or edit a new one
3. Add a section of type `Skills`
4. Add a title
5. Add an item inside with an image and filling all required fields
6. Click Save
7. Image disapears (But is saved on database and API returns it)

### Expected behavior

The image field should keep filled with my selected image.

I have another section with the same type of structure with an image and it behaves normally, looking for the difference between then I notice that the only diference is that in `Skills` I called the repeatable section as `items`, and the other section that works I called with `social_media`. 

Renaming the `items` field inside `Skills` for any other name, makes my component behaves as expected. 

Long story short, when I have a repeatable component called `items` with an media field inside, strapi resets that field on save.

### Screenshots

![image](https://user-images.githubusercontent.com/6539258/146796481-3ae5dee0-7c4a-4395-ad71-bae7e3a20351.png)

![image](https://user-images.githubusercontent.com/6539258/146796535-87f57fe3-7538-4e8c-918a-397f7507bb35.png)

![image](https://user-images.githubusercontent.com/6539258/146796576-78bc5315-8651-4e10-8442-2abddec9a0f6.png)

### System

- Node.js version: v16.13.1
- NPM version: 8.1.2
- Strapi version: 4.0.0
- Database: PostgreSQL
- Operating system: Windows with Ubuntu on WSL

**Comment by MarcosGin** [comment]
I encountered this same error when we have a component at level 3.

`LandingPage (component) -> QuoteSlider (component) ->  Items -> (Component) -> media property`

Reviewing the request to the backend a bit, it can be seen that at that level of iteration, the properties are not returned as multimedia
![image](https://user-images.githubusercontent.com/26859121/147160857-29137b3e-2827-4381-aa74-16c26e373a81.png)

There is some fix for this?

**Comment by willianrod** [comment]
@MarcosGin as a workaround I changed the name of the array "items". In my case it seems to make the behavior go back to normal. Check if it works for in your case too

**Comment by kasonde** [comment]
Unable to reproduce in version 4.0.0 or in the latest version. Works as expected.

**Comment by talazenkot** [comment]
This still happens in 4.1.3

**Comment by derrickmehaffy** [comment]
Confirmed based on user reports and duplicate issues

**Comment by pepijn-vanvlaanderen** [comment]
The patch from seems unrelated to the original issue: https://github.com/strapi/strapi/issues/11319#issuecomment-1026717218

I had the same issue as @willianrod, but then with a component named `cards` not `items`. We also got it fixed by renaming the component to `card_items`.

**Comment by dogunbound** [comment]
Can confirm this is still a problem on a clean strapi 4.1.7 project.

**Comment by dogunbound** [comment]
Similar issue to: https://github.com/strapi/strapi/issues/11909

Video explaining: https://github.com/strapi/strapi/issues/11955video explaining why I think so: https://www.youtube.com/watch?v=j-TYly0s0L0

**Comment by derrickmehaffy** [comment]
Confirmed via an EE customer:

![image](https://user-images.githubusercontent.com/8593673/165853817-712db3cf-e8f9-4e0a-8896-853d5d42f15f.png)
![image](https://user-images.githubusercontent.com/8593673/165853880-c8b3af6f-0af6-4cf0-a4e9-8df2a5c0a52a.png)

on v4.1.8

**Comment by raimonds-wrong** [comment]
+1; Strapi v.4.1.8. This seems to be due to a repeatable field name. Common structure: dynamic zone component > repeatable > image field. If the name of a repeatable field matches in several dynamic zone components, for example – named 'items', the image disappears in the editor view when saved, but is available in the API data.

I have encountered this problem several times because it is not easily noticeable. But can be corrected by renaming the repeatable field otherwise. Also have to add extra populate fields in API request. 😞

**Comment by raimonds-wrong** [comment]
Just updated to v4.1.9. Deleted .cache and rebuild admin UI. The problem persists ➕

**Comment by thbwd** [comment]
Please note that we provided a minimal, fully reproducible example of this bug in https://github.com/strapi/strapi/issues/12930. This really is a deal-breaker.

**Comment by dogunbound** [comment]
> Please note that we provided a minimal, fully reproducible example of this bug in #12930. This really is a deal-breaker.

There is a workaround. Watch this video: https://www.youtube.com/watch?v=j-TYly0s0L0

**Comment by Peretyaka** [comment]
STRAPI VERSION
v4.1.11
NODE VERSION
v14.18.3
SQLite

Problem exist.

**Comment by petersg83** [comment]
Hello everyone :) Thank you for your inputs, I could spend some time on this and I confirm what @raimonds-wrong and @dogunbound say:
**The problem comes from the fact that in your dynamic zone, you have at least 2 different components that each have a field with the same name => This field is then badly populated because there is a mix between the two that is done.**

The faulty code is here : https://github.com/strapi/strapi/blob/master/packages/core/utils/lib/convert-query-params.js#L178 : the schema of the different components are merged.

I don't have a solution yet, we need to think about it

**Comment by DenisLaboureyras** [comment]
Hello @petersg83 ,

Very interesting ! I've run into this issue, as well as another one that seems related so I wanted to share it with you.
When you configure fields with the same name in dynamic zones and different attributes, the graphql schema is not responding correctly. For example, I've configured two fields named title, one required and one not required, graphql expect both fields to be required and gave me an error if the second title was not filled. The workaround for me was to configure both fields not required...

Bottom line, I think this faulty code may cause other bugs.

Good luke to find the solution :)

**Comment by WiktorG** [comment]
Any update on this guys? :D

**Comment by Convly** [comment]
Hey there, @petersg83 is currently on holiday, so let's let him rest under the palm trees :palm_tree: 

As for the issue above, it indeed all comes from the fact we're generating a fake DynamicZone schema with attributes from all the possible components being merged altogether. If some share the same name, only the last one will be kept.
[This comment](https://github.com/strapi/strapi/blob/master/packages/core/utils/lib/convert-query-params.js#L170-L171) explains why we had to do it like this: Basically, we did that to fix the deep filtering on DZ, but for that, we needed a way to know on which attribute it was possible to do it. It's far from a perfect solution ([context](https://github.com/strapi/strapi/pull/12579#discussion_r811936592)), but fixing this issue means implementing a brand new way of querying the DynamicZones without exploding the number of requests made to the database. It's doable but it's rather complex in theory and we did not have time to dedicate to this in the past.

If you've any other idea to share, please don't hesitate :eyes: 

TBH, I don't really have any magic solution yet but it could be interesting to try and add this refactoring in an upcoming sprint initiative for our Content or Developer-Experience squad.

pinging @marcoautiero @yanniskadiri to make them aware of that issue :slightly_smiling_face:

**Comment by boxcc** [comment]
from #13894 
I found that the data fetched from the API was incomplete

![WeChatd82c90f1ca7b0b8f6cd1ac7d98f9de54](https://user-images.githubusercontent.com/11386943/181727305-cf957ae5-c295-4daf-aad4-5987bcbf1bf9.png)

**Comment by boxcc** [comment]
I found https://github.com/strapi/strapi/blob/21b95f9954a3ad622ff42dfdb34172bee7483284/packages/core/utils/lib/convert-query-params.js#L173-L179 have same key bug. because "Feature row group" and "Feature columns group" both have field "features", the "Feature" of "Feature columns group" will be omitted

**Comment by Convly** [comment]
I believe https://github.com/strapi/strapi/pull/13855 should have fixed this one.

**Comment by deschantkn** [comment]
I am currently experiencing a similar issue that is messing up my development flow. Every time I make a change to any models in the content-type builder, the data in all my repeatable components seem to just vanish.

Upon trying to re-enter the lost data, I get the following internal server error

```
error: update "components_homepage_features" set "id" = $1, "description" = $2, "title" = $3, "link_name" = $4 where ("id" = $5) - column "description" of relation "components_homepage_features" does not exist
error: update "components_homepage_features" set "id" = $1, "description" = $2, "title" = $3, "link_name" = $4 where ("id" = $5) - column "description" of relation "components_homepage_features" does not exist
```

I then have to go back to the component to delete and add back every field before I can save data successfully. See the components below:

<details>
<summary>src/components/homepage/feature.json</summary>

```json
{
  "collectionName": "components_homepage_features",
  "info": {
    "displayName": "Feature",
    "icon": "star",
    "description": ""
  },
  "options": {},
  "attributes": {
    "description": {
      "type": "text",
      "required": true
    },
    "title": {
      "type": "string",
      "required": true
    },
    "linkName": {
      "type": "string",
      "required": true
    }
  }
}

```

</details>

<details>
<summary>src/components/homepage/features.json</summary>

```json
{
  "collectionName": "components_homepage_features",
  "info": {
    "displayName": "Features",
    "icon": "award",
    "description": ""
  },
  "options": {},
  "attributes": {
    "featureList": {
      "type": "component",
      "repeatable": true,
      "component": "homepage.feature",
      "required": true
    },
    "title1": {
      "type": "string",
      "required": true
    },
    "title2": {
      "type": "string",
      "required": true
    },
    "buttonName": {
      "type": "string",
      "required": true
    }
  }
}

```

</details>

#### System

- Node.js version: v16.15.0
- NPM version: 8.5.5
- Strapi version: 4.4.5
- Database: PostgreSQL 14
- Operating system: macOS Monterey Version 12.6


#### Additional context

I would like to mention that the project is in TypeScript and was initialized with pnpm using the following command `pnpm dlx create-strapi-app@latest`. 

I installed the dependencies with Yarn version 1.22.19 


If this isn't related to this issue then I am happy to create a bug report with the standard format and more details. Thanks.
